src/client/datascience/types.ts

```diff
@@ -937,7 +936,7 @@ export interface ICellHashProvider {
     getHashes(): IFileHashes[];
     getExecutionCount(): number;
     incExecutionCount(): void;
-    generateHashFileName(cell: ICell, executionCount: number): string;
+    onBeforeCellExecute(notebookCell: NotebookCell): Promise<void>;
```

Comment:
I'd rename this to `addCellHash` or something.
We know why we're calling this method, its to generate a hash. Hvaing a method named `onBeforeCellExecute` means its just an event handler & we have no idea what its doing.

I.e. i'd prefer us to be more explicit about the action we're performing. Else this is similar to a logger, kind of a black box, but that's not the case (i.e. we know what this class is meant to be used for).